### DISCUSSION TITLE
Inserting multiple rows from Rust

### DISCUSSION THREAD

**Discussion by alexpdp7** [cozodb/cozo#249] [Q&A]
Hi,

I'm writing a Rust program that parses a Markdown file, extracting tables and code blocks.

The tables should be inserted into a database. Then, the code blocks contain queries over those tables.

If I have a vector of headers, and a vector of rows (where every row is a vector of values)... am I supposed to convert and feed those into [`import_relations`](https://docs.rs/cozo/latest/cozo/struct.Db.html#method.import_relations)? Or should I iterate over rows and use `run_script` to run queries to insert row by row?

Cheers,

Álex

**Comment by aramallo** [comment]
Hi @alexpdp7 importing relations should be faster but the downside is that it will not trigger index creation nor custom triggers. If you have indices over the relation you want to import data to you have to either build a script string or manually create the indices later through another script AFAIK.
